llvm/lib/CodeGen/AsmPrinter/AsmPrinter.cpp

```diff
@@ -1326,16 +1326,24 @@ static unsigned getBBAddrMapMetadata(const MachineBasicBlock &MBB) {
          (const_cast<MachineBasicBlock &>(MBB).canFallThrough() << 3);
 }
 
-void emitYkBBAddrMapSymbol(const MachineFunction &MF, MCStreamer &OutStreamer,
+/// Emit a start (or stop) marker symbol into the `.llvm_bb_addr_map` section
+/// so that we can find the extent of the section at runtime.
+///
+/// The `MCStreamer` should be primed to output to the `.llvm_bb_addr_map`
+/// section prior to calling this function.
+///
+/// This assumes that LTO is being used (as is required for the Yk JIT), and
+/// thus that there is only a single `Module` in play, and in turn that no
+/// symbol clashes can occur.
+void emitYkBBAddrMapSymbol(MCContext &MCtxt, MCStreamer &OutStreamer,
                            bool Start) {
-  std::string SymName("ykllvm.bbaddrmap.");
-  SymName.append(MF.getName().str());
+  std::string SymName("ykllvm.bbaddrmaps.");
   if (Start)
-    SymName.append(".start");
```

Comment:
Silly question maybe: why have we changed `.start` to `start`?

Reply:
If you remove the `.` from line 1340 instead, the diff becomes smaller.

Reply:
Fixed in [93a89e6](https://github.com/ykjit/ykllvm/pull/49/commits/93a89e6363a1255e642f8422076015e43890edc7).